src/reflection.cpp

```diff
@@ -481,4 +481,232 @@ Offset<const Table *> CopyTable(FlatBufferBuilder &fbb,
   }
 }
 
+bool VerifyStruct(flatbuffers::Verifier &v,
+                  const flatbuffers::Table &parentTable,
+                  voffset_t fieldOffset,
+                  const reflection::Object &obj,
+                  bool isRequired) {
+  assert(obj.is_struct());
+  auto offset = parentTable.GetOptionalFieldOffset(fieldOffset);
+  if (!isRequired) {
```

Comment:
check for required centrally (i.e. where you're iterating thru the fields), which would remove the code duplication between the if/else here and below.